synapseclient/models/user.py

```diff
@@ -0,0 +1,311 @@
+        display_name: This field is deprecated and will always be null.
+        r_studio_url: URL for RStudio server assigned to the user.
+        summary: A summary description about this person.
+        position: This person's current position title.
+        location: This person's location.
+        industry: The industry/discipline that this person is associated with.
+        company: This person's current affiliation.
+        profile_picure_file_handle_id: The File Handle id of the user's profile picture.
+        url: A link to more information about this person.
+        team_name: This person's default team name.
+        notification_settings: Contains a user's notification settings.
+        preferences: User preferences
+        created_on: The date this profile was created.
+    """
+
+    id: Optional[int] = None
+    """A foreign key to the ID of the 'principal' object for the user."""
+
+    username: Optional[str] = None
+    """A name chosen by the user that uniquely identifies them."""
+
+    etag: Optional[str] = None
+    """Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle
+    concurrent updates. Since the E-Tag changes every time an entity is updated it is
+    used to detect when a client's current representation of an entity is out-of-date."""
+
+    first_name: Optional[str] = None
+    """This person's given name (forename)"""
+
+    last_name: Optional[str] = None
+    """This person's family name (surname)"""
+
+    emails: List[str] = field(default_factory=list)
+    """The list of user email addresses registered to this user."""
+
+    open_ids: List[str] = field(default_factory=list)
+    """The list of OpenIds bound to this user's account."""
+
+    r_studio_url: Optional[str] = None
+    """URL for RStudio server assigned to the user"""
+
+    summary: Optional[str] = None
+    """A summary description about this person"""
+
+    position: Optional[str] = None
+    """This person's current position title"""
+
+    location: Optional[str] = None
+    """This person's location"""
+
+    industry: Optional[str] = None
+    """The industry/discipline that this person is associated with"""
+
+    company: Optional[str] = None
+    """This person's current affiliation"""
+
+    profile_picure_file_handle_id: Optional[str] = None
+    """The File Handle id of the user's profile picture"""
+
+    url: Optional[str] = None
+    """A link to more information about this person"""
+
+    team_name: Optional[str] = None
+    """This person's default team name"""
+
+    send_email_notifications: Optional[bool] = True
+    """Should the user receive email notifications? Default true."""
+
+    mark_emailed_messages_as_read: Optional[bool] = False
+    """Should messages that are emailed to the user be marked as
+    READ in Synapse? Default false."""
+
+    preferences: Optional[List[UserPreference]] = field(default_factory=list)
```

Comment:
Nit: Not related, but I don't understand this `preferences` parameter.  I took a look at the API and it doesn't tell you much about it either...

Reply:
I was messing around with this as well since it isn't explained anywhere. It let's you set arbitrary booleans on your account. For example if you get the userprofile for my account i set an example value. `bfauble`